### PR TITLE
Bobcat: allow setting default profile

### PR DESCRIPTION
**Pull request by dolik-rce**
Continuation of #22.

There are two main changes from user perspective:
1. User can set default profile, current default is indicated by "D" in the profile icon.
2. Currently active profile is indicated by bold text.

I believe that explicitly marking profile as default is more intuitive than to force user to always put his default profile on top. As an example, let's assume that someone would like to keep his profiles alphabetically sorted, for better orientation, but his default default profile would not be first in the alphabetical order. Then the user would have to either choose different name or make exception to his preferred ordering.

### CODE REVIEW

**Comment by dolik-rce** [comment]
TODO:
- [ ] fix behavior when there is no default profile (e.g. when it is deleted, or before any profile is created)

**Comment by ismail-yilmaz** [comment]
In general, there is nothing I would object to about these set of patches. They are as they should be (great). But they overlook one point: Active terminal might not always be present. On the command line `bobcat -s|--settings` can run the settings dialog without running bobcat and  this is intentional.

File: Bobcat/Profile.cpp [hunk 1]
```diff
@@ -433,6 +433,9 @@ int Profiles::Load()
 		for(const auto& p : ~profiles)
 			list.Add(p.key, RawToValue(clone(p.value)));
 		list.Sort(1, [](const Value& a, const Value& b) -> int { return a.To<Profile>().order - b.To<Profile>().order; });
+		int current = list.Find(ctx.GetActiveTerminal()->profilename, 0);
```
Comment (ismail-yilmaz) [inline Bobcat/Profile.cpp:436]: Active terminal may not be available (Settings dialog can be run without the app), and bobcat will crash.

File: Bobcat/Profile.cpp [hunk 2]
```diff
@@ -12,21 +12,12 @@ namespace Upp {
...
-
-struct NormalProfileNameDisplayCls : Display
-{
-	void Paint(Draw& w, const Rect& r, const Value& q, Color ink, Color paper, dword style) const final
-	{
-		StdDisplay().Paint(w, r, AttrText(q).SetImage(Images::Terminal()), ink, paper, style);
-	}
-};
+	const Image& img = ctx.settings.defaultprofile == q ? Images::DefaultTerminal() : Images::Terminal();
+	bool current = ctx.GetActiveTerminal()->profilename == q;
```
Comment (ismail-yilmaz) [inline Bobcat/Profile.cpp:18]: Again, active terminal may not be present, GetActiveTerminal may return nullptr and then dialog will crash.

**Comment by dolik-rce** [comment]
> Active terminal might not always be present.

You're right, I overlooked that. I'll fix it when I get back from weekend trip.

File: Bobcat/Profile.cpp (hunk 1 above)
Comment (dolik-rce) [reply Bobcat/Profile.cpp]: Should be fixed now.

File: Bobcat/Profile.cpp (hunk 2 above)
Comment (dolik-rce) [reply Bobcat/Profile.cpp]: Ditto...

**Comment by ismail-yilmaz** [comment]
Looks good

**Comment by ismail-yilmaz** [comment]
@dolik-rce 

By the way, there is one thing I modified with your code: ProfileNameDisplay. I hope you don't mind that. I can understand your reason to move its declaration to the header file, but I prefer to keep them tight in their source files. Instead, Bobcat is now derived from the `Upp::Pte` class (i.e. it now has a smart pointer) and I also introduced `GetContext()` global function. This function can be called from any global function to check if Bobcat is existing or destroyed and then the context can be accessed. As you can guess, this is also the first step towards a "lite" plugin infrastructure for Bobcat.

**Pull request merged**